### DISCUSSION TITLE
Updating Positron via winget

### DISCUSSION THREAD

**Discussion by dj-m** [posit-dev/positron#5178] [Q&A] [answered]
Requesting advice from the Positron gods on here.

When I look on the manifest for [winget](https://github.com/microsoft/winget-pkgs/tree/master/manifests/p/Posit/Positron), Positron has a 2024.11.0 version. However, when I run 
```
winget show Posit.Positron --versions
```
there isn't anything later than the 2024.09.0 version. 

Am I missing something fairly obvious here  or does it need a [wingetcreate](https://learn.microsoft.com/en-us/shows/open-at-microsoft/wingetcreate-keeping-winget-packages-up-to-date) update process solution? 

Thanks in advance!

**Comment by cderv** [comment, marked as answer]
I know there is a cache logic in winget. This is what I get on my side
```
❯ winget show Posit.Positron --versions
Trouvé Positron [Posit.Positron]
Version
---------
2024.11.0
2024.10.0
2024.09.0
2024.08.0
2024.07.0
```

2024.11.0 is in the list. 

You could try maybe `winget source update` to update the source for your local winget, and see if this change.
The source for winget is a cached one somehow, through a CDN so it needs update before you can see it.
````
❯ winget source list
Nom     Argument                                      Contenu explicite
-----------------------------------------------------------------------
msstore https://storeedgefd.dsx.mp.microsoft.com/v9.0 false
winget  https://cdn.winget.microsoft.com/cache        false
```` 

I don't remember exactly how often winget to update its cache for source, but this has happen in the past. 
- https://github.com/posit-dev/positron/issues/3859#issuecomment-2239978919
- https://github.com/posit-dev/positron/issues/3859#issuecomment-2240373478

**Reply by dj-m** [reply]
Hi @cderv, thanks for the assist!

Unfortunately the ```winget source update``` didn't address the issue right now but at least we know this is a local-ish issue. 

I'll close the discussion since the resolution is already out there, as you've linked to. Thanks again.
